File: src/Domain/Entities/Appointment.cs
using Domain.ValueObjects;

namespace Domain.Entities;

public class Appointment
{
    public Guid Id { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string? CustomerPhone { get; set; }
    public string? CustomerEmail { get; set; }
    public DateTimeOffset Start { get; set; }
    public DateTimeOffset End { get; set; }
    public string? Notes { get; set; }
    public string? ExternalId { get; set; }

    public TimeRange ToTimeRange() => new(Start, End);
}

File: client/src/types/appointment.ts
export interface Appointment {
  id: string;
  customerName: string;
  customerPhone: string | null;
  customerEmail: string | null;
  start: string;
  end: string;
  notes: string | null;
  externalId: string | null;
}

export interface CreateAppointmentPayload {
  customerName: string;
  customerPhone: string | null;
  customerEmail: string | null;
  start: string;
  end: string;
  notes: string | null;
}